raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -463,9 +464,14 @@ public void transitionToCandidate() throws IOException {
         return state;
     }
 
-    private void transitionTo(EpochState state) throws IOException {
+    private void transitionTo(EpochState state) {
         if (this.state != null) {
-            this.state.close();
+            try {
+                this.state.close();
+            } catch (IOException e) {
+                throw new UncheckedIOException(
+                        "Failed to close Quorum status during transition to candidate, state is " + this.state, e);
```

Comment:
How about "Failed to transition from ${this.state.name()} to ${state.name()}"? Note that this string is  not valid Java.